src/test/scala/ConsumerTestSuite.scala

```diff
@@ -0,0 +1,165 @@
+      override protected val consumer: MockConsumer[String, String] = mockConsumer
+    }
+
+    testConsumer.assign(topicInfoList)
+
+    val topics = mockConsumer.assignment().asScala.map(_.topic())
+
+    assert(infoList.map(_.topic).toSet == topics)
+  }
+
+  "assign" should "throw Exception" in {
+    val topic1 = "topic1"
+    val topic2 = "topic2"
+    val infoList: List[TopicInfo] = List(
+      TopicInfo(topic = topic1),
+      TopicInfo(topic = topic2)
+    )
+    val topicInfoList = TopicInfoList(infoList)
+
+    val mockConsumer = new MockConsumer[String, String](OffsetResetStrategy.EARLIEST)
+
+    val testConsumer = new Consumer[String, String]("127.0.0.1:9000", "groupId", pollTimeout, enableAutoCommit, autoCommitInterval) {
+      override protected val consumer: MockConsumer[String, String] = mockConsumer
+    }
+
+    assertThrows[Exception](testConsumer.assign(topicInfoList))
+  }
+
+  "assign" should "assign to all partitions in topics with offset" in {
+    val mockConsumer = new MockConsumer[String, String](OffsetResetStrategy.EARLIEST)
+    val testConsumer = new Consumer[String, String]("127.0.0.1:9000", "groupId", pollTimeout, enableAutoCommit, autoCommitInterval) {
+      override protected val consumer: MockConsumer[String, String] = mockConsumer
+    }
+
+    val firstTopicPartition = TopicPartitionInfo(topic = "topic1", partition = 0, offset = 1)
+    val secondTopicPartition = TopicPartitionInfo(topic = "topic1", partition = 1, offset = 2)
+    val thirdTopicPartition = TopicPartitionInfo(topic = "topic2", partition = 2, offset = 3)
+
+    val chechpointInfoList: TopicPartitionInfoList = TopicPartitionInfoList(
+      List(firstTopicPartition, secondTopicPartition, thirdTopicPartition)
+    )
+
+    testConsumer.assign(chechpointInfoList)
+
+    val topicPartitions = mockConsumer.assignment().asScala.toList
+    val offsets = topicPartitions.map { x =>
+      mockConsumer.position(x)
+    }
+    assert(offsets.toSet == chechpointInfoList.entities.map(_.offset).toSet)
+  }
+
+  "poll" should "return ConsumerRecord" in {
+    val mockConsumer = new MockConsumer[String, String](OffsetResetStrategy.EARLIEST)
+    val testConsumer = new Consumer[String, String]("127.0.0.1:9000", "groupId", pollTimeout, enableAutoCommit, autoCommitInterval) {
+      override protected val consumer: MockConsumer[String, String] = mockConsumer
+    }
+
+    val firstTopicPartition = TopicPartitionInfo(topic = "topic1", partition = 0, offset = 1)
+
+    val chechpointInfoList: TopicPartitionInfoList = TopicPartitionInfoList(
+      List(firstTopicPartition)
+    )
+    val expectedRecord = new ConsumerRecord[String, String]("topic1", 0, 1, "key", "value")
+
+    testConsumer.assign(chechpointInfoList)
+    mockConsumer.addRecord(expectedRecord)
+
+    val actualRecord = testConsumer.poll().asScala.toList.head
+
+    assert(actualRecord == expectedRecord)
+  }
+
+  "commit" should "commit offset for topic partition" in {
```

Comment:
add one more TopicPartitionInfo and commit several CheckpointInfo

Reply:
ok